### ISSUE TITLE
Google map Reverse Geocoding

### ISSUE THREAD

**Issue by ruscon** [geocoder-php/Geocoder#546]
https://developers.google.com/maps/documentation/geocoding/intro#ReverseGeocoding
These are the optional parameters: result_type and  location_type - can we add them ?

**Comment by arubacao** [comment]
@Nyholm I'd also like this kind of functionality and I'm ready to send a PR for it.

Just some things to ask you upfront: 

1. `result_type` and `location_type` can be set through the constructor

```
public function __construct(HttpAdapterInterface $adapter, $locale = null, $region = null, $useSsl = false, $apiKey = null, $result_type = null, $location_type = null)
```
(as last parameters for backwards compatibility)

2.  `result_type` and `location_type` can be set through setters

3. Extend `buildQuery` method

3. Create new Exception class `InvalidRequestException` and throw it on `status` = `INVALID_REQUEST` responses

Please add your thoughts and comments & tell me if you approve this approach.
I'd appreciate a quick answer, so I can get started fast.

Cheers,
Chris

**Comment by Nyholm** [comment]
This is now added. Have a look at #619 and #620

**Comment by Nyholm** [comment]
Closed by #619 and #620
